site/source/docs/contributing/developers_guide.rst

```diff
@@ -104,37 +104,54 @@ Bisecting
 If you find a regression, bisection is often the fastest way to figure out what
 went wrong. This is true not just for finding an actual regression in Emscripten
 but also if your project stopped working when you upgrade, and you need to
-investigate if it's an Emscripten regression or something else.
+investigate if it's an Emscripten regression or something else. The rest of
+this section covers bisection on Emscripten itself.
 
-The normal ``git bisect`` workflow can work if you just need to bisect a single
-repository. For example, you can bisect only on the emscripten repo if you are on
-a range that all works with the same versions of LLVM and Binaryen (which was
-mentioned earlier, are two separate codebases that are depended on).
+If you have a large bisection range - for example, that covers more than one
+version of Emscripten - then you probably have changes across multiple repos
+(Emscripten, LLVM, and Binaryen). In that case the easiest and fastest thing
+is to bisect using **emsdk builds**. Each step of the bisection will download
+a "tip of tree" (tot) build from the emsdk. Using this approach you don't need
+to compile anything yourself, so it can be very fast!
 
-If you have a large bisection range, you generally can't bisect a single repo.
-You can still bisect, though! To do that you need the emsdk and to understand
-how the
+To do this, you need a basic understanding of Emscripten's
 `release process <https://github.com/emscripten-core/emscripten/blob/main/docs/process.md#release-processes>`_
-works for all the repos together. The key "trick" is that::
-
-     emsdk install tot
-
-can install an arbitrary build of emscripten: it installs the one identified
-in ``emscripten-releases-tot.txt``. You can therefore bisect on the
-`releases repo <https://chromium.googlesource.com/emscripten-releases>`_ which
-has a DEPS file that specifies what version of all the various repos are
-in which release. At each bisection step in this repo the git hash identifies a
-particular `tot` release (which when it was built, was tip-of-tree).
-
-For the actual bisection you can use
-`git bisect <https://git-scm.com/docs/git-bisect>`_. Each step in will download
-a complete build which is not a trivial download. However, at least the number
-of such steps will be logarithmic!
-
-This bisects down to a single commit in the releases repo. That commit will
-generally update a single sub-repo from one commit to another. That will often
-be a very short list or even a single commit. If it's more than one, you can
-bisect there while using a fixed build for the other repos.
+The key idea is that::
+
+     emsdk install HASH
+
+can install an arbitrary build of emscripten from any point in the history
+of the emsdk. Each build is identified by a hash (a long string of numbers
```

Comment:
"any point in the history of emscripten-releases (assuming the build succeeded)."